diff --git a/benchmarks/dynamo/common.py b/benchmarks/dynamo/common.py
index c00501c0166..1c68dfbb133 100644
--- a/benchmarks/dynamo/common.py
+++ b/benchmarks/dynamo/common.py
@@ -689,7 +689,18 @@ def speedup_experiment(args, model_iter_fn, model, example_inputs, **kwargs):
             frozen_model_iter_fn = export_aot_inductor(model_iter_fn)
         else:
             frozen_model_iter_fn = torch._dynamo.run(model_iter_fn)
-
+        
+        optimized_model = copy.deepcopy(model)
+        fusion_path = os.environ["FUSION_PATH"]
+        if self.args.backend == 'ipex':
+            import intel_extension_for_pytorch as ipex
+
+            if self.args.bfloat16:
+                optimized_model = ipex.optimize(optimized_model, dtype=torch.bfloat16, weights_prepack=False)
+            else:
+                optimized_model = ipex.optimize(optimized_model, weights_prepack=False)
+            
+            ipex._set_compiler_backend(fusion_path) # jit path: "torchscript", inductor path: "inductor".
         for rep in trange(args.repeat, desc="running benchmark"):
             inputs = (
                 randomize_input(copy.deepcopy(example_inputs))
@@ -717,7 +728,7 @@ def speedup_experiment(args, model_iter_fn, model, example_inputs, **kwargs):
 
             with maybe_mark_profile(p=p, mark="actual"):
                 timings[rep, 1], actual_output = timed(
-                    model,
+                    optimized_model,
                     frozen_model_iter_fn,
                     inputs,
                     return_result=True,
